test/Shared/UserManagerTestBase.cs

```diff
@@ -506,27 +499,25 @@ protected TRole CreateRole(string namePrefix = "")
             IdentityResultAssert.IsSuccess(await manager.CreateAsync(user, "password"));
             var result = await manager.ChangePasswordAsync(user, "bogus", "newpassword");
             IdentityResultAssert.IsFailure(result, "Incorrect password.");
-            IdentityResultAssert.VerifyUserManagerFailureLog(manager.Logger, "ChangePasswordAsync", user.Id.ToString(), IdentityErrorDescriber.Default.PasswordMismatch());
+            IdentityResultAssert.VerifyUserManagerFailureLog(manager.Logger, "ChangePasswordAsync", await manager.GetUserIdAsync(user), IdentityErrorDescriber.Default.PasswordMismatch());
         }
 
         [Fact]
         public async Task AddDupeUserNameFails()
         {
             var manager = CreateManager();
-            var user = CreateTestUser();
-            var user2 = new TUser() { UserName = user.UserName };
+            var user = CreateTestUser("AddDupeUserNameFails");
+            var user2 = CreateTestUser("AddDupeUserNameFails");
```

Comment:
So if CreateTestUser only accepts name prefixes, just use SetuserName to explicitly set it to a fixed name after creating the user.